app/repositories/streams_repository.rb

```diff
@@ -17,4 +17,11 @@ def calc_average_value!(stream)
     stream.average_value = stream.measurements.average(:value)
     stream.save!
   end
+
+  def add_start_coordinates!(stream)
+      first_measurement = stream.measurements.order(time: :asc).first
```

Comment:
This is only for mobile sessions? In fixed we have a bug where sometimes the measurements' times are wrong (1970 or 2010). I think that is the affected column. Maybe you could double check with Radek if it's best to order by time or id. Also, if this is slow, depending on the indexes ordering by time or id could make a difference

Reply:
good catch! I'll check that.

Reply:
Unfortunately, using id's is not a good idea. We don't sort the measurements before saving, so the order might be random. Also if we would sort them it would be by time anyway :/ But we have an index on time, so at least this shouldn't influence the performance much.
So I'm going to use time but this is only executed for Mobile, so I guess it's ok

Reply:
We should try to make it clear what's the path for Mobile and what's the path for Fixed. Right now reading that code is a struggle. (The code that was there already I mean)